### ISSUE TITLE
A bug

### ISSUE THREAD

**Issue by 11051911** [Marigoldwu/A-Unified-Framework-for-Deep-Attribute-Graph-Clustering#9]
When I brought my own data into the experiment, it basically ran out, but the methods involving cross-entropy loss reported this error. May I ask the author what the problem may be? Have you encountered it before?
`C:\cb\pytorch_1000000000000\work\aten\src\ATen\native\cuda\Loss.cu:106: block: [32,0,0], thread: [126,0,0] Assertion `target_val >= zero && target_val <= one` failed.
C:\cb\pytorch_1000000000000\work\aten\src\ATen\native\cuda\Loss.cu:106: block: [4255,0,0], thread: [62,0,0] Assertion `target_val >= zero && target_val <= one` failed. 
  File "E:\最新聚类论文代码\A-Unified-Framework-for-Deep-Attribute-Graph-Clustering-main\main.py", line 53, in <module>
    result = train(args, data, logger)
  File "E:\最新聚类论文代码\A-Unified-Framework-for-Deep-Attribute-Graph-Clustering-main\model\pretrain_gat_for_efrdgc\train.py", line 48, in train
    loss = F.binary_cross_entropy(A_pred.view(-1), adj_label.view(-1))
`

**Comment by Marigoldwu** [comment]
> When I brought my own data into the experiment, it basically ran out, but the methods involving cross-entropy loss reported this error. May I ask the author what the problem may be? Have you encountered it before? `C:\cb\pytorch_1000000000000\work\aten\src\ATen\native\cuda\Loss.cu:106: block: [32,0,0], thread: [126,0,0] Assertion `target_val >= zero && target_val <= one`failed. C:\cb\pytorch_1000000000000\work\aten\src\ATen\native\cuda\Loss.cu:106: block: [4255,0,0], thread: [62,0,0] Assertion`target_val >= zero && target_val <= one`failed. File "E:\最新聚类论文代码\A-Unified-Framework-for-Deep-Attribute-Graph-Clustering-main\main.py", line 53, in <module> result = train(args, data, logger) File "E:\最新聚类论文代码\A-Unified-Framework-for-Deep-Attribute-Graph-Clustering-main\model\pretrain_gat_for_efrdgc\train.py", line 48, in train loss = F.binary_cross_entropy(A_pred.view(-1), adj_label.view(-1))`

@11051911 Hello, thank you for your question. I have indeed encountered similar problems before, and it often happens with new models or new data sets. I think the possible reason is that during forward propagation, the values of some tensors may be NaN or INF. You can debug to see where the calculation went wrong and try to do special processing for these values, such as replacing NaN with 0 or replacing INF with 1 (as appropriate). If the above method cannot solve the problem, please feel free to continue the discussion. My WeChat: 18751766925.
